jax/_src/numpy/lax_numpy.py

```diff
@@ -2673,11 +2673,35 @@ def wrapper(*args, **kwargs):
     return tuple(asarray(x) for x in f(*args, **kwargs))
   return wrapper
 
-tril_indices = _wrap_indices_function(np.tril_indices)
-triu_indices = _wrap_indices_function(np.triu_indices)
 mask_indices = _wrap_indices_function(np.mask_indices)
 
 
+def _triu_size(n, m, k):
+  if k < 0:
+    return n * m - _triu_size(m, n, (1 - k))
+  elif k >= m:
+    return 0
+  else:
+    mk = _min(n, m - k)
+    return mk * (mk + 1) // 2 + mk * (m - k - mk)
+
+
+@_wraps(np.triu_indices)
+def triu_indices(n, k, m=None):
```

Comment:
`k` defaults to 0 per numpy. Also, maybe add type hint and/or jit compile?

Reply:
Thanks - fixed in #14364

I don't think it's all that important to JIT compile here, particularly since every argument is static.